### PR TITLE
Interactive Time Travel Queries

### PR DESCRIPTION
**Pull request by qianl15**
This PR describes how to perform interactive time travel queries through `psql` and DBOS time travel proxy.

### CODE REVIEW

File: docs/cloud-tutorials/interactive-timetravel.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+Let's assume you've deployed the ["Hello, Database" quickstart](../getting-started/quickstart) application to DBOS Cloud.
+The application's `dbos_hello` table tracks how many times each person has been greeted.
+The following query tells you how many times Mike has been greeted:
+
+```sql
+postgres=> select greet_count from dbos_hello where name = 'Mike';
+ greet_count
+-------------
+           8
+```
```
Comment (maxdml) [inline docs/cloud-tutorials/interactive-timetravel.md:58]: This query be done from the `dbos_hello`. `postgres` won't have the relations in the common case.

File: docs/cloud-tutorials/interactive-timetravel.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+postgres=> select greet_count from dbos_hello where name = 'Mike';
+ greet_count
+-------------
+           4
+```
+
+You can run any `SELECT` statement on the database to query its state as of the timestamp you chose.
+Statements that modify schemas or data (`INSERT`, `UPDATE`, `DROP TABLE`, etc.) will not have any effect.
+At any time, you can run `DBOS TS <timestamp>` again to travel to a different time.
+You can also run `DBOS SNAPSHOT RESET` to return to the present time.
```
Comment (maxdml) [inline docs/cloud-tutorials/interactive-timetravel.md:76]:
```suggestion
You can also run `DBOS SNAPSHOT RESET;` to return to the present time.
```

File: docs/cloud-tutorials/interactive-timetravel.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+           8
+```
+
+Now, let's time travel!
+To view your database at a past point in time, you can set the timestamp through the special `DBOS TS <timestamp>` command.
+We support any timestamp string in [RFC 3339 format](https://datatracker.ietf.org/doc/html/rfc3339).
+For example, to view your database at 4:00:00 PM UTC on 2024-04-26, and see how many times Mike had been greeted as of then, run:
+
+```sql
+postgres=> DBOS TS '2024-04-26T16:00:00';
```
Comment (maxdml) [inline docs/cloud-tutorials/interactive-timetravel.md:66]: needs TZ, e.g., `'DBOS TS "2024-04-24T08:30:00-07:00';`

File: docs/cloud-tutorials/interactive-timetravel.md (hunk 1 above)
Comment (kraftp) [reply docs/cloud-tutorials/interactive-timetravel.md:58]: This is just the way the proxy displays it

File: docs/cloud-tutorials/timetravel-debugging.md [hunk 4]
```diff
@@ -195,10 +195,10 @@ For non-VS Code users, you can run the time-travel debugger manually through the
 The time travel debugger requires our debug proxy to transform database queries so that it can "travel" back in time.
 You can download the pre-compiled debug proxy using the following link. Please choose the one based on your operating system and hardware platform:
 
-- [Download for macOS (Intel Chip)](https://dbos-releases.s3.us-east-2.amazonaws.com/debug-proxy/1.0.2/debug-proxy-macos-x64-1.0.2.zip)
-- [Download for macOS (Apple Chip)](https://dbos-releases.s3.us-east-2.amazonaws.com/debug-proxy/1.0.2/debug-proxy-macos-arm64-1.0.2.zip)
-- [Download for Linux (x86_64)](https://dbos-releases.s3.us-east-2.amazonaws.com/debug-proxy/1.0.2/debug-proxy-linux-x64-1.0.2.zip)
-- [Download for Linux (arm)](https://dbos-releases.s3.us-east-2.amazonaws.com/debug-proxy/1.0.2/debug-proxy-linux-arm64-1.0.2.zip)
+- [Download for macOS (Intel Chip)](https://dbos-releases.s3.us-east-2.amazonaws.com/debug-proxy/1.1.7/debug-proxy-macos-x64-1.1.7.zip)
```
Comment (devhawk) [inline docs/cloud-tutorials/timetravel-debugging.md:198]: This is out of date. 1.2.10 is latest release. Is there a way we can automate updating this?

Comment (kraftp) [reply docs/cloud-tutorials/timetravel-debugging.md:198]: Yes--one way to do it is to create a "latest release" object in the bucket and have your scripts publish to it as well as to the appropriate version.

**Review by kraftp — APPROVED** [review]

**Pull request merged**